xapian-core/include/xapian/weight.h

```diff
@@ -372,6 +395,10 @@ class XAPIAN_VISIBILITY_DEFAULT Weight {
     Xapian::termcount get_wdf_upper_bound() const {
 	return wdf_upper_bound_;
     }
+
+    static void parameter_error(const char * param, const std::string & scheme);
+
+    bool double_param(const char ** p, double * ptr_val) const;
```

Comment:
> Neither seems to use the context of a Weight object - could they both be made static and added to Weight::Internal instead?

I tried that but build fails at weighting scheme classes seemingly due to the dependency between `create_from_parameters` in Weight subclasses and `parameter_error` in Weight::Internal which, since builds after Weight subclasses, causes these errors:

```bash
weight/bb2weight.cc:190:10: error: incomplete type ‘Xapian::Weight::Internal’ used in nested name specifier
     if (!Xapian::Weight::Internal::double_param(&p, &k))
          ^
weight/bb2weight.cc:191:2: error: incomplete type ‘Xapian::Weight::Internal’ used in nested name specifier
  Xapian::Weight::Internal::parameter_error("Parameter is invalid", "bb2");
  ^
weight/bb2weight.cc:193:2: error: incomplete type ‘Xapian::Weight::Internal’ used in nested name specifier
  Xapian::Weight::Internal::parameter_error("Extra data after parameter", "bb2");
```
What do you suggest?